### PR TITLE
chore(markdown): Removed old static markdown library in favor of compose...

### PR DESCRIPTION
**Pull request by Srokap**
...r version. Deprecated old Markdown function.

### CODE REVIEW

File: engine/lib/plugins.php [hunk 1]
```diff
@@ -522,8 +522,6 @@ function _elgg_plugins_init() {
 	elgg_register_action('admin/plugins/deactivate_all', '', 'admin');
 
 	elgg_register_action('admin/plugins/set_priority', '', 'admin');
-
-	elgg_register_library('elgg:markdown', elgg_get_root_path() . 'vendors/markdown/markdown.php');
```
Comment (jdalsem) [inline engine/lib/plugins.php]: Is pointing it to an empty lib just as BC breaking? Why not leave it intact but not use it anymore, and remove it in 2.0?

Comment (mrclay) [inline engine/lib/plugins.php]: Consider a 3rd-party plugin that calls `elgg_load_library('elgg:markdown')`. if you leave the lib as is, PHP will fatal as the old lib tries to redefine markdown(). If you remove the lib, Elgg will exception out.

**Comment by jdalsem** [comment]
I think this fixes #5408

**Comment by Srokap** [comment]
It's now rebased and updated.

File: composer.json [hunk 2]
```diff
@@ -16,7 +16,8 @@
         "knplabs/gaufrette": "~0.1.0",
         "tedivm/stash": "~0.12",
         "ircmaxell/password-compat": "~1.0",
-        "roave/security-advisories": "dev-master"
+        "roave/security-advisories": "dev-master",
+        "michelf/php-markdown": "1.4.1"
```
Comment (mrclay) [inline composer.json]: Why not ~1.4.1?

File: engine/lib/deprecated-1.10.php [hunk 3]
```diff
@@ -53,3 +53,14 @@ function generate_user_password(\ElggUser $user, $password) {
...
 }
+
+
+/**
+ * Convert Markdown to HTML.
+ *
+ * @deprecated 1.10
+ */
+function Markdown($text) {
+	elgg_deprecated_notice(__FUNCTION__ . ' is deprecated. Use \Michelf\MarkdownExtra::defaultTransform()', "1.10");
```
Comment (mrclay) [inline engine/lib/deprecated-1.10.php]: ".. is deprecated. Please provide your own Markdown implementation." Also add the second sentence after `@deprecated 1.10 `

**Comment by mrclay** [comment]
Please resubmit for 1.x.

**Pull request closed without merging**